src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1310,6 +1318,13 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         if (!CheckHWIntrinsicImmRange(intrinsic, simdBaseJitType, immOp1, mustExpand, immLowerBound, immUpperBound,
                                       hasFullRangeImm, &useFallback))
         {
+            // If it needs a fallback - we have two options: Importer-level via impNonConstFallback and lower-level
+            if (useFallback && !HWIntrinsicInfo::NoJmpTableImm(intrinsic) &&
+                !HWIntrinsicInfo::MaybeNoJmpTableImm(intrinsic))
+            {
+                *pImportAsCall = true;
```

Comment:
I'm not following where the logic that allows the call to be converted back into an intrinsic is?

That is, I'm not following the logic that allows this:
> The interesting side-effect from this is that we now can inline must-expand SIMD intrsinsics which we give up due to non-constant args for IMM etc. It allows jit to fold the fallback if late phases detect a constant

I would've expected either we import the `GT_HWINTRINSIC` node and rewrite it back to a `GT_CALL` in a late phase (such as rationalization). Or we instead import as `GT_CALL` and then have a later phase (morph and/or rationalization) attempt to resolve the tracked method handled into an intrinsic again (repeating the checks that the right input is a constant).